ui/src/app/machines/views/MachineDetails/MachineNetwork/BridgeForm/BridgeForm.tsx

```diff
@@ -15,14 +55,94 @@ const BridgeForm = ({
   selected,
   systemId,
 }: Props): JSX.Element | null => {
+  const dispatch = useDispatch();
+  const machine = useSelector((state: RootState) =>
+    machineSelectors.getById(state, systemId)
+  );
+  const cleanup = useCallback(() => machineActions.cleanup(), []);
+  const nextName = getNextNicName(machine, NetworkInterfaceTypes.BRIDGE);
+  const [{ linkId, nicId }] = selected;
+  const nic = useSelector((state: RootState) =>
+    machineSelectors.getInterfaceById(state, systemId, nicId, linkId)
+  );
+  const vlan = useSelector((state: RootState) =>
+    vlanSelectors.getById(state, nic?.vlan_id)
+  );
+  const vlansLoading = useSelector(vlanSelectors.loading);
+  const { errors, saved, saving } = useMachineDetailsForm(
+    systemId,
+    "creatingBridge",
+    "createBridge",
+    () => close()
+  );
+
+  useEffect(() => {
+    dispatch(vlanActions.fetch());
+  }, [dispatch]);
+
   // A bridge can only be created with one interface.
   if (selected.length !== 1) {
     return null;
   }
+
+  if (vlansLoading || !nic || !machine || !("interfaces" in machine)) {
+    return <Spinner text="Loading..." />;
+  }
+
   return (
     <FormCard sidebar={false} stacked title="Create bridge">
       <InterfaceFormTable interfaces={selected} systemId={systemId} />
-      <button onClick={close}>Cancel</button>
+      <FormikForm
+        buttons={FormCardButtons}
+        cleanup={cleanup}
+        errors={errors}
+        initialValues={{
+          ...networkFieldsInitialValues,
+          bridge_fd: 15,
+          bridge_stp: false,
+          bridge_type: BridgeType.STANDARD,
+          // Prefill the fabric from the parent interface.
+          fabric: vlan?.fabric,
+          mac_address: "",
+          name: nextName,
+          tags: [],
+          // Prefill the vlan from the parent interface.
+          vlan: nic.vlan_id,
+        }}
+        onSaveAnalytics={{
+          action: "Create bridge",
+          category: "Machine details networking",
+          label: "Create bridge form",
+        }}
+        onCancel={close}
+        onSubmit={(values: BridgeFormValues) => {
+          // Clear the errors from the previous submission.
+          dispatch(cleanup());
+          type Payload = BridgeFormValues & {
+            parents: NetworkInterface["parents"];
+            system_id: Machine["system_id"];
+          };
+          const payload: Payload = {
+            ...values,
+            parents: [nic.id],
+            system_id: systemId,
+          };
+          // Remove all empty values.
+          Object.entries(payload).forEach(([key, value]) => {
```

Comment:
Should we turn this into a util? Maybe something that can be given a list of exceptions when `""` is a valid value.

Reply:
Good idea, I've filed https://github.com/canonical-web-and-design/maas-ui/issues/2334.